src/compile.rs

```diff
@@ -421,6 +421,75 @@ pub fn validate_rules(statements: &BTreeSet<ValidatedStatement>) -> Result<(), C
     errors.into_result(())
 }
 
+// Go through all functions and check if they are castable
+// based only on their args
+pub fn initialize_castable(functions: &mut FunctionMap, types: &TypeMap) {
```

Comment:
This looks like it's checking the declared arguments in the function signature.  Is the relevant thing to check the arguments passed in a call?

Reply:
We are marking some functions as non castable with this method so I am not sure if it is 100% useless.